addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxServerHandler.java

```diff
@@ -839,14 +869,134 @@ public void updateListener(SqueezeBoxPlayerEventListener listener) {
                 } else if (function.equals("volume")) {
                     final int volume = (int) Double.parseDouble(value);
                     updatePlayer(new PlayerUpdateEvent() {
+
                         @Override
                         public void updateListener(SqueezeBoxPlayerEventListener listener) {
                             listener.absoluteVolumeChangeEvent(mac, volume);
                         }
+
                     });
                 }
             }
         }
+
+        private void handleFavorites(String message) {
+            logger.trace("Handle favorites message: {}", message);
+
+            String[] messageParts = message.split("\\s");
+            if (messageParts.length == 2 && "changed".equals(messageParts[1])) {
+                // LMS informing us that favorites have changed; request an update to the favorites list
+                requestFavorites();
+                return;
+            }
+            if (messageParts.length < 7) {
+                logger.trace("No favorites in message.");
+                return;
+            }
+
+            List<Favorite> favorites = new ArrayList<>();
+            String title = "";
+            String count = "";
+            Favorite f = null;
+            for (String part : messageParts) {
+                String id;
```

Comment:
Why do you need all these fields a quick looks suggest that you either assign them immediately or discard the data, wdyt?

Reply:
Same as my comment above. They are returned in the response from the LMS, even though they currently are not used.